### PR TITLE
Feat/upstream rewrite

### PR DESCRIPTION
**Pull request by DanielRailean**
This PR allows the deployment of multiple Lambda endpoints as part of a single Kong service, by introducing the ability to override service upstream. 

This change allows the plugin to be used on a per-route basis and thus have multiple lambda routes that together form a service.

# About the PR

## Changelog
- Add: Added 3 configuration options, to make possible target override.
- Add: Used those configuration options in the access phase of the plugin as proxy target if enabled.
- Update: Updated developer readme.
- Note: Both changes above are fully backward compatible.

### Testing
- [x] Has been tested locally and against a deployed lambda.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -4,10 +4,6 @@ on:
   pull_request:
     branches:
       - main
-    paths:
-      - 'kong/**/*.lua'
-      - 'spec/*.lua'
-
```
Comment (Nicoolai) [inline .github/workflows/build.yml:10]: Why remove this?
By removing the Lua paths, it means that the linting workflow will run even if only markdown files are changed.

Comment (DanielRailean) [reply .github/workflows/build.yml]: It wasn't running as it was, just updated it to include only `spec` and `kong`

File: Readme.md [hunk 2]
```diff
@@ -29,13 +29,35 @@ aws_region - AWS region where your Lambda is deployed to
...
+
+override_target_host - To be used when deploying multiple lambdas on a single Kong service (because lambdas have differennt URLs)
+type = "string"
+required = false
+
+override_target_port - To be used when deploying a lambda on a Kong service that listens on a port other than `443`
+type = "number"
+required = false
+
+override_target_protocol - To be used when deployinglambda on a Kong service that has a protocol different than `https`
```
Comment (alexpopaconst) [inline Readme.md:44]:
```suggestion
override_target_protocol - To be used when deploying lambda on a Kong service that has a protocol different than `https`
```

File: Readme.md [hunk 3]
```diff
@@ -29,13 +29,35 @@ aws_region - AWS region where your Lambda is deployed to
...
+required = false
 ```
 
-### AWS Setup required
+## Service vs Route scoped configuration
+
+The plugin can be enabled on a per service as well as on a per route basis.
+
+When enabling the plugin on a service/route, the plugin will use the service upstream as the lambda target, unless `override_target_host` is specified.
+This configuration works fine only if a single Lambda is used in the entire service, if multiple lambdas are specified on a per path basis, without the `override_target_host` configured, they will all use the service upstream as the target, which will obviously fail.
```
Comment (alexpopaconst) [inline Readme.md:55]: maybe not say that it will "obviously fail",  but explain that having the same host will fail at the lambda level, since it expects a different host.

File: Readme.md [hunk 4]
```diff
@@ -29,13 +29,35 @@ aws_region - AWS region where your Lambda is deployed to
...
+
+***TLDR: If multiple lambdas are required for a single Kong service (which most likely is the case), the correct configuration is having a Kong route for each Lambda, enabling the plugin o a per-route basis with `override_target_host` on each of them, so the requests are routed to the right lambda.***
```
Comment (alexpopaconst) [inline Readme.md:57]:
```suggestion
***TLDR: If multiple lambdas are required for a single Kong service, the correct configuration is having a Kong route for each Lambda, enabling the plugin o a per-route basis with `override_target_host` on each of them, so the requests are routed to the right lambda.***
```

**Review by alexpopaconst — CHANGES_REQUESTED** [review]
